src/features/abi-methods/data/abi-method.ts

```diff
@@ -0,0 +1,255 @@
+import { Atom, atom } from 'jotai'
+import { getApplicationsAppSpecsAtom } from '@/features/abi-methods/data/index'
+import algosdk, { ABIReferenceType, TransactionType } from 'algosdk'
+import { uint8ArrayToBase64 } from '@/utils/uint8-array-to-base64'
+import { TransactionResult } from '@algorandfoundation/algokit-utils/types/indexer'
+import { Round } from '@/features/blocks/data/types'
+import { AppSpecVersion } from '@/features/abi-methods/data/types'
+import { getGroupResultAtom } from '@/features/groups/data'
+import { TransactionId } from '@/features/transactions/data/types'
+import { base64ToBytes } from '@/utils/base64-to-bytes'
+import { AbiMethod, AbiMethodArgument, AbiMethodReturn, AbiValue, AbiType } from '@/features/abi-methods/models'
+
+export const abiMethodResolver = (transaction: TransactionResult): Atom<Promise<AbiMethod | undefined>> => {
+  return atom(async (get) => {
+    if (!isValidAppCallTransaction(transaction)) {
+      return undefined
+    }
+
+    const abiMethod = await get(getAbiMethodAtom(transaction))
+    if (!abiMethod) return undefined
+
+    const methodArguments = await get(getMethodArgumentsAtom(transaction, abiMethod))
+    const methodReturn = getMethodReturn(transaction, abiMethod)
+
+    return {
+      name: abiMethod.name,
+      arguments: methodArguments,
+      return: methodReturn,
+    } satisfies AbiMethod
+  })
+}
+
+const getAbiMethodAtom = (transaction: TransactionResult): Atom<Promise<algosdk.ABIMethod | undefined>> => {
+  return atom(async (get) => {
+    if (!isValidAppCallTransaction(transaction)) {
```

Comment:
I tend to keep the guard on top the function regardless if the caller already check it so that if this function (`getAbiMethodAtom`) is called somewhere, the guard makes sure that it behaves as expected.

Reply:
As discussed, I removed the guards and use invariants